src/list.cpp

```diff
@@ -160,15 +160,21 @@ void list_probes(const BPFtrace &bpftrace, const std::string &search_input)
       executable = executable.substr(0, executable.find(":"));
 
       auto paths = resolve_binary_path(executable);
-      if (paths.size() > 1)
+      switch (paths.size())
       {
-        std::cerr << "path '" << executable << "' must refer to a unique binary but matched " << paths.size() << std::endl;
+      case 0:
+        std::cerr << "uprobe target '" << executable << "' does not exist or is not executable" << std::endl;
         return;
-      }
-      absolute_exe = paths.front();
-      struct stat s;
-      if (stat(absolute_exe.c_str(), &s) != 0) {
-        std::cerr << "uprobe target " << executable << " does not exist" << std::endl;
+      case 1:
+        absolute_exe = paths.front();
+        struct stat s;
+        if (stat(absolute_exe.c_str(), &s) != 0) {
```

Comment:
Do we still need this check here? `resolve_binary_path` already does an `access` check.